www/docs/reactjs/mocking.md

```diff
@@ -0,0 +1,222 @@
+
+:::
+
+### 2. Use your handlers and mocks to simulate an API call
+
+Let's say that we recently defined an endpoint in our `cats` router to serve everyone with information about our favorite cat.
+
+<details><summary>Backend code</summary>
+
+```ts title='server/routers/cats.ts'
+export const catsRouter = router({
+  bestCat: publicProcedure
+    .input(z.object({ name: z.string() }))
+    .query(({ input }) => {
+      return {
+        bestCat: `Hey ${input.name}! The best cat is Mozzie!`,
+      };
+    }),
+});
+```
+
+</details>
+
+In addition, we've also created a simple React component, `BestCat` that displays the return of this query.
+
+<details><summary>Frontend code</summary>
+
+```ts title='component/cats/BestCat.tsx'
+export default const BestCat = () => {
+  const { loading, error, data } = trpc.cats.bestCat.useQuery({ name: "John" })
+  if (loading || error) return <p>Data unavaliable</p>;
+  return (
+    <StyledCatText>{data.bestCat}</StyledCatText>
+  )
+}
+```
+
+</details>
+
+Finally, we want to verify that our component is working as we anticipated, so we write a test.
+
+```ts title='component/cats/BestCat.spec.tsx'
+describe("BestCat", () => {
+  const server = setupServer(
+    trpcMsw.cats.bestCat.query((req, res, ctx) => {
+      const input = req.getInput()
+      // 🚨 You can console.log here to view your request being captured
+      return res(
+        ctx.status(200),
+        ctx.data({
+          bestCat: `Hey ${input.name}! The best cat is Mozzie!`,
+        })
+      );
+    })
+  );
+  
+  beforeAll(() => server.listen());
+  afterAll(() => server.close());
+  
+  test("should render the result of our query", async () => {
+    renderWithProviders(<BestCat />);
+    expect(screen.getByText(/Data unavaliable/i)).toBeVisible();
+    await waitFor(() => {
+      const worldsBestCat = screen.getByText(/Mozzie/i);
+      expect(worldsBestCat).toBeVisible();
+    });
+  });
+});
+```
+
+Let's break our test down a little to understand what's going on. We'll start by looking at what our `test` block is doing, and then return to our `server` setup and teardown.
+
+The very first thing we do is make sure that our client (`trpc`) has access to its context (`MockedTRPCProvider`). Happily, we've made this easy to do by utilizing our new `renderWithProviders` function. After that, we are correctly anticipating that when we first render the component, we need to asynchronously wait for the data to load before we can assert on the anticipated response.
+
+The component is now successfully making calls from its client when we run the test, however, we now need to intercept those calls and mock a response. This is where our `server` comes in. Inside of `setupServer`, we can define the requests that our `Cat` component uses so we can intercept them and return a mocked response. Because `trpcMsw` uses the shape of our `AppRouter`, we can use our existing typing to quickly define the request handler we need.
```

Comment:
```suggestion
The component is now successfully making calls from its tRPC client when being tested. But we don't want it to make those calls to a real server. This is where our mocked `server` comes in. Inside of `setupServer`, we can define the requests that our `Cat` component uses so we can intercept them and return a mocked response. Because `trpcMsw` uses the shape of our `AppRouter`, we can use our existing typing to quickly define the request handler we need.
```